core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -805,56 +835,47 @@ boolean computeIfPresent(K key, Consumer<OakWriteBuffer> computer) {
             throw new NullPointerException();
         }
 
-        int infiLoopCount = 0;
-        while (true) {
-            infiLoopCount++;
-            if (infiLoopCount > 1000) {
-                System.out.println("Stuck in non-zc get");
-                assert false;
-            }
+        ThreadContext ctx = getThreadLocalContext();
+
+        for (int i = 0; i < MAX_RETRIES; i++) {
             Chunk<K, V> c = findChunk(key); // find chunk matching key
-            EntrySet.LookUp lookUp = c.lookUp(key);
-            if (lookUp == null || lookUp.valueSlice == null) {
+            c.lookUp(ctx, key);
+            if (!ctx.isValueValid()) {
                 return null;
             }
 
-            if (updateVersionAfterLinking(c, lookUp)) {
+            if (updateVersionAfterLinking(c, ctx)) {
                 continue;
             }
-            Result<T> res = valueOperator.transform(lookUp.valueSlice, transformer, lookUp.version);
+            Result res = valueOperator.transform(ctx.result, ctx.value, transformer);
             if (res.operationResult == RETRY) {
                 continue;
             }
-            return res.value;
+            return (T) res.value;
         }
 
+        throw new RuntimeException("getValueTransformation failed: reached retry limit (1024).");
     }
 
     <T> T getKeyTransformation(K key, OakTransformer<T> transformer) {
-        ByteBuffer serializedKey = getKey(key);
-        if (serializedKey == null) {
-            return null;
-        }
-        return transformer.apply(serializedKey.slice().asReadOnlyBuffer());
-    }
-
-    // Returns the buffer of key and not value as usual
-    private ByteBuffer getKey(K key) {
         if (key == null) {
             throw new NullPointerException();
         }
 
+        ThreadContext ctx = getThreadLocalContext();
         Chunk<K, V> c = findChunk(key);
-        EntrySet.LookUp lookUp = c.lookUp(key);
-        if (lookUp == null || lookUp.valueSlice == null) {
+        c.lookUp(ctx, key);
+        if (!ctx.isValueValid()) {
             return null;
         }
-        return c.readKeyFromEntryIndex(lookUp.entryIndex);
+        return transformer.apply(ctx.key.getDuplicatedReadByteBuffer().slice());
     }
 
     ByteBuffer getMinKey() {
```

Comment:
Although getMin/MaxKey is currently not in use, but generally should return OakDetachedReadBuffer...

Reply:
Changed.